src/guides/v2.3/test/testing.md

```diff
@@ -87,6 +87,103 @@ They are executed during continuous integration using the `bin/magento` tool.
 
 Please see the [`magento dev:tests:run`][] documentation for more information, using the test type `static`.
 
+### PHPStan
+
+[PHPStan][] is a static analysis tool for PHP. Magento now integrates PHPStan by default.
+It is available for developers to use in their own work, located at `Magento\Test\Php\LiveCodeTest::testPhpStan()`.
+
+It is run on changed `*.php` files using rule strictness [level 0][] and detects issues such as:
+
+-  Method should return something but return statement is missing.
+-  Access to constant on an unknown class
+-  Access to undefined constant
+-  Access to an undefined property
+-  Anonymous function has an unused use
+-  Array has N duplicate keys
+-  Call to an undefined method
+
+#### Suppressing errors
+
+To ignore errors in a particular piece of code, use the `phpstan:ignore` annotation.
+
+```php
+// phpstan:ignore "Error message"
+```
+
+The error message text is optional.
+It is possible to use the `*` wildcard within messages.
+
+```php
+/**
+ * Class ClassWithIgnoreAnnotation
+ */
+class ClassWithIgnoreAnnotation
+{
+    /**
+     * Test method.
+     * phpstan:ignore "Method level error"
+     */
+    public function getProductList()
+    {
+        // phpstan:ignore "Method Magento\PhpStan\Formatters\Fixtures\ClassWithIgnoreAnnotation::testMethod() invoked with 2 parameters, 1 required."
+        $this->testMethod('test1', 'test2');
+        // phpstan:ignore "Method * invoked with 2 parameters, 1 required."
+        $this->testMethod('test1', 'test2');
+        // phpstan:ignore
+        $this->testMethod('test1', 'test2');
+        $this->testMethod('test1', 'test2'); // phpstan:ignore
+    }
+    /**
+     * @param string $arg1
+     * @return string
+     */
+    private function testMethod(string $arg1)
+    {
+        return $arg1;
+    }
+}
+```
+
+#### Adding files/directories to PHPStan blacklist
+
+If some files are broken on purpose, add them to the blacklist to ignore them.
+
+The blacklist is located at `dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/blacklist/`.
+
+```config
+lib/internal/Magento/Framework/Interception/Test/Unit/Config/ConfigTest.php
+lib/internal/Magento/Framework/Cache/Backend/Eaccelerator.php
+```
+
+#### Exclude error messages from the analysis results
+
+If some error messages are not relevant for the current testing scenario, use regular expressions to filter them.
+
+The PHPStan configuration file is `dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/phpstan.neon`.
+
+Example:
+
+```config
+ignoreErrors:
+    # Ignore PHPStan\Rules\Classes\UnusedConstructorParametersRule
+   - '#Constructor of class [a-zA-Z0-9\\_]+ has an unused parameter#'
+```
+
+#### Exclude directories from analysis using the `excludes_analyse` option
+
+As needed in some cases, directories and files can be excluded via the PHPStan configuration file:
```

Comment:
Via is on a poo list somewhere.
https://wiki.corp.magento.com/display/MDOC/Style+guide+-+grammar#Styleguide-grammar-Wording

```suggestion
As needed in some cases, you can exclude specific directories and files using the `excludes_analyse` section in the PHPStan configuration file:
```